topi/recipe/gemm/gemm_int8.py

```diff
@@ -0,0 +1,174 @@
+"Example code to perform int8 GEMM"
+import logging
+import sys
+import numpy as np
+import tvm
+from tvm import autotvm
+
+
+def intrin_dot():
+    n = 4  # dp4a requires operands packed by 4
+    x = tvm.placeholder((n,), name='x', dtype='int8')
+    y = tvm.placeholder((n,), name='y', dtype='int8')
+    k = tvm.reduce_axis((0, n), name='k')
+
+    z = tvm.compute(
+        (1,), lambda _: tvm.sum(
+            x[k].astype('int32') * y[k].astype('int32'), axis=k))
+
+    def intrin_func(ins, outs):
+        xx, yy = ins
+        zz = outs[0]
+        ib = tvm.ir_builder.create()
+
+        dp4a = zz.vstore(0, tvm.call_pure_extern('int32', '__dp4a',
+                                                 xx.vload(0, dtype='int8x4'),
+                                                 yy.vload(0, dtype='int8x4'),
+                                                 zz.vload(0)))
+        ib.emit(dp4a)
+
+        body = ib.get()
+        return body, zz.vstore(0, 0), body
+
+    with tvm.build_config(data_alignment=4, offset_factor=1) as cfg:
+        binds = {t: tvm.decl_buffer(t.shape, t.dtype, t.op.name,
+                                    data_alignment=cfg.data_alignment,
+                                    offset_factor=cfg.offset_factor,
+                                    scope='local') for t in [x, y, z]}
+        return tvm.decl_tensor_intrin(z.op, intrin_func, binds=binds)
+
+
+@autotvm.template
+def gemm_int8(n, m, l):
+    A = tvm.placeholder((n, l), name='A', dtype='int8')
+    B = tvm.placeholder((m, l), name='B', dtype='int8')
+
+    k = tvm.reduce_axis((0, l), name='k')
+    C = tvm.compute((n, m), lambda i, j: tvm.sum(A[i, k].astype('int32') * B[j, k].astype(
+        'int32'), axis=k), name='C')
+
+    cfg = autotvm.get_config()
+    s = tvm.create_schedule(C.op)
+    y, x = C.op.axis
+
+    AA = s.cache_read(A, 'shared', [C])
+    BB = s.cache_read(B, 'shared', [C])
+    AL = s.cache_read(AA, 'local', [C])
+    BL = s.cache_read(BB, 'local', [C])
+    CC = s.cache_write(C, 'local')
+
+    dot = intrin_dot()
```

Comment:
Move this declaration out of template. This can accelerate the feature extraction in tuning